src/solver/simulation/include/antares/solver/simulation/solver.hxx

```diff
@@ -489,9 +489,16 @@ void ISimulation<ImplementationType>::regenerateTimeSeries(uint year)
         if (refreshTSonCurrentYear)
         {
             auto clusters = getAllClustersToGen(study.areas, pData.haveToRefreshTSThermal);
-            fs::path savePath = fs::path(study.folderOutput.to<std::string>()) / "ts-generator"
-                                / "thermal" / "mc-" / std::to_string(year);
-            generateThermalTimeSeries(study, clusters, savePath.string());
+            generateThermalTimeSeries(study, clusters, study.runtime->random[Data::seedTsGenThermal]);
+
+            bool archive = study.parameters.timeSeriesToArchive & Data::timeSeriesThermal;
```

Comment:
Details about if we print the TS or not were moved here (to the solver). 
Indeed, these details do not belong to the TS generation functions (as they used to) : TS generator executable generate and prints, and don't want to know if study or user wants to print or not.